### PR TITLE
chore(e2e): Update dependency to eventing v0.12.0

### PR DESCRIPTION
**Pull request by rhuss**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -23,7 +23,7 @@ require (
 	knative.dev/eventing v0.12.0
 	knative.dev/pkg v0.0.0-20200122022923-4e81bc3c320f
 	knative.dev/serving v0.12.0
-	knative.dev/test-infra v0.0.0-20200116044902-d5990f0e5a05
+	knative.dev/test-infra v0.0.0-20200129221128-ee08b33b3cf0
```
Comment (navidshaikh) [inline go.mod]: rechecked the change went into test-infra: https://github.com/knative/test-infra/pull/1620/files
the filename change applies to nightly as well as release YAML. 
So even with commit `d5990f0e5a05` pinned in client's go.mod, this should've worked?

/cc: @chizhg @adrcunha

Comment (navidshaikh) [reply go.mod]: The error in CI is:
```
error: unable to read URL "https://storage.googleapis.com/knative-releases/eventing/previous/v0.12.0/release.yaml", server reported 404 Not Found, status code=404
```

this worked for nightly release of eventing, but release YAML filename still appears to be populated as in URL `release.yaml`

Comment (adrcunha) [reply go.mod]: @chizhg looks like test-infra needs to be updated:

https://github.com/knative/test-infra/blob/8f405dd7f685816dee00efe945f18d2e105ef46b/scripts/library.sh#L461

Comment (chizhg) [reply go.mod]: Filed https://github.com/knative/test-infra/pull/1650..
Please update test-infra here after the PR is merged @navidshaikh

Comment (navidshaikh) [reply go.mod]: Thanks!

The PR is merged!
/cc: @rhuss

**Pull request merged**